three-cards-divination/src/test/java/com/cyber/mysticism/tarot/service/UserServiceTest.java

```diff
@@ -0,0 +1,124 @@
+package com.cyber.mysticism.tarot.service;
+
+import com.cyber.mysticism.tarot.model.TarotUser;
+import com.cyber.mysticism.tarot.repository.UserRepository;
+import com.cyber.mysticism.tarot.service.exceptions.UserAlreadyExistsException;
+import com.cyber.mysticism.tarot.service.exceptions.UserNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+    @InjectMocks
+    private UserService userService;
+
+    private TarotUser tarotUser;
+
+    @BeforeEach
+    void setup() {
+        tarotUser = new TarotUser();
+        tarotUser.setUsername("user");
+        tarotUser.setEmail("email");
+    }
```

Comment:
**Suggestion:** It's better to use `@BeforeEach` to setup common test data to avoid code duplication. [best practice]
```suggestion
    @BeforeEach
    void setup() {
        tarotUser = new TarotUser();
        tarotUser.setUsername("user");
        tarotUser.setEmail("email");
    }
```